src/vivarium_gates_mncnh/data/lbwsg_rr_caps/generate_caps.py

```diff
@@ -0,0 +1,207 @@
+"""
+LBWSG RR Caps Generator
+-----------------------
+
+This script generates capped relative risk values for low birth weight and short gestation.
+
+Usage:
+    python generate_caps.py [-l LOCATION] [-o OUTPUT_DIR]
+
+Arguments:
+    -l, --location     The location for which to generate LBWSG RR caps.
+                      Example: Ethiopia, Nigeria, Pakistan
+                      Default: Ethiopia
+
+    -o, --output-dir   The output directory where the generated CSV file will be saved.
+                      Default: /ihme/homes/hjafari/artifacts/test_caps/
+
+Example:
+    python generate_caps.py -l Nigeria -o /path/to/output_dir
+
+The output will be a CSV file named <location>.csv in the specified output directory.
+"""
+
+import argparse
+
+import numpy as np
+import pandas as pd
+from scipy.optimize import minimize
+
+from vivarium_gates_mncnh.constants import data_keys, data_values
+from vivarium_gates_mncnh.data.loader import get_data, load_standard_data
+
+
+def prepare_rr_cap_inputs(
+    rrs: pd.DataFrame, location: str
+) -> dict[tuple[str, str], dict[str, pd.DataFrame]]:
+    """
+    Returns a dictionary mapping (sex, age_group) to a dict with keys 'exposure', 'rrs', and 'acmrisk',
+    each containing a pd.DataFrame.
+    """
+    data: dict[tuple[str, str], dict[str, pd.DataFrame]] = {}
+    for sex in ["Male", "Female"]:
+        # Early neonatal
+        data[(sex, "early_neonatal")] = {
+            "exposure": get_data(data_keys.LBWSG.EXPOSURE, location).query(
+                f"child_age_start==0 & sex_of_child=='{sex}'"
+            ),
+            "rrs": rrs.query(f"age_start == 0 & sex=='{sex}'"),
+            "acmrisk": get_data(
+                data_keys.POPULATION.ALL_CAUSES_MORTALITY_RISK, location
+            ).query(f"age_start == 0 and sex=='{sex}'"),
+        }
+        # Late neonatal
+        data[(sex, "late_neonatal")] = {
+            "exposure": get_data(data_keys.LBWSG.EXPOSURE, location).query(
+                f"child_age_start==0.01917808 & sex_of_child=='{sex}'"
+            ),
+            "rrs": rrs.query(f"age_start == 0.01917808 & sex=='{sex}'"),
+            "acmrisk": get_data(
+                data_keys.POPULATION.ALL_CAUSES_MORTALITY_RISK, location
+            ).query(f"age_start==0.01917808 & sex=='{sex}'"),
+        }
+    return data
+
+
+def calculate_maximum_mr_by_draw(
+    input_data: dict[tuple[str, str], dict[str, pd.DataFrame]],
+    rr_cap: float,
+    draw: int,
+    sex: str,
+    age_group: str,
+) -> pd.DataFrame:
+    exposure = input_data[(sex, age_group)]["exposure"]
+    rrs = input_data[(sex, age_group)]["rrs"]
+    acmrisk = input_data[(sex, age_group)]["acmrisk"]
+
+    # rename index columns to be consistent with other indices
```

Comment:
Everything below this is copied and pasted from [this research notebook](https://github.com/ihmeuw/vivarium_research_mncnh_portfolio/blob/main/data_prep/lbwsg_rr_caps.ipynb), except that location column is no longer present in these data.